src/main/java/ch/njol/skript/sections/EffSecShoot.java

```diff
@@ -190,14 +190,11 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 		direction = (Expression<Direction>) exprs[3];
 
 		if (sectionNode != null) {
-			AtomicBoolean delayed = new AtomicBoolean(false);
-			Runnable afterLoading = () -> delayed.set(!getParser().getHasDelayBefore().isFalse());
-			trigger = loadCode(sectionNode, "shoot", afterLoading, ShootEvent.class);
-			if (delayed.get()) {
-				Skript.error("Delays cannot be used within a 'shoot' effect section");
-				return false;
-			}
+			trigger = SectionUtils.loadLinkedCode("shoot", (beforeLoading, afterLoading)
```

Comment:
thoughts about adding a helper method that just has the node, name and event as arguments? that way it ensures nobody will call the wrong method without the runnables

Reply:
We aren't able to call the methods that build the triggers (e.g. `loadCode`) as they have restricted access (and it should probably remain that way). Unless you mean something else